rust/arrow/src/datatypes.rs

```diff
@@ -357,10 +373,12 @@ impl JsonSerializable for u16 {
 }
 
 impl ArrowNativeType for u16 {
+    #[inline]
     fn from_usize(v: usize) -> Option<Self> {
         num::FromPrimitive::from_usize(v)
     }
 
+    #[inline]
```

Comment:
Do those help for performance? AFAIK inline has impact mostly across crates?  I would expect most simple functions to be inlined in Arrow itself already.

Reply:
Yep, it took me 3hs to find out why there was an 100% degradation without inlines. :(

My findings is that the compiler is being unable to optimize certain operations inside iterators when they are not `inlined`.

I even copy-pasted `memory.rs`, `bytes.rs`, `buffer.rs` and `bit_utils` to a separate repo/lib to compare the performance on the same code, and the only difference between then was `usize::try_from(u32)` vs `u32.to_usize` (via `ArrowNativeType`).

I think that we should re-visit all these functions in `ArrowNativeType`. It is not clear why they should be there.